packages/commonwealth/client/scripts/views/pages/CommunityManagement/Integrations/Discord/Discord.tsx

```diff
@@ -57,16 +68,45 @@ const Discord = () => {
           redirect_domain: isCustomDomain ? window.location.origin : undefined,
         }),
       );
-      const link = `https://discord.com/oauth2/authorize?client_id=${process.env.DISCORD_CLIENT_ID}&permissions=1024&scope=applications.commands%20bot&redirect_uri=${redirectURL}/discord-callback&response_type=code&scope=bot&state=${currentState}`;
+      const link =
+        `https://discord.com/oauth2/authorize?client_id=${process.env.DISCORD_CLIENT_ID}` +
+        `&permissions=1024&scope=applications.commands%20bot&redirect_uri=${redirectURL}` +
+        `/discord-callback&response_type=code&scope=bot&state=${currentState}`;
       window.open(link, '_parent');
 
       setConnectionStatus('none');
     } catch (e) {
-      notifyError('Failed to connect discord!');
+      notifyError('Failed to connect Discord!');
       setConnectionStatus('none');
     }
   }, [connectionStatus]);
 
+  const onDisconnect = useCallback(async () => {
+    if (connectionStatus === 'connecting' || connectionStatus === 'none')
+      return;
+    try {
+      await axios.post(`${app.serverUrl()}/removeDiscordBotConfig`, {
+        community_id: app.activeChainId(),
+        jwt: app.user.jwt,
+      });
+      // remove channel query from cache to force refresh
+      queryClient.removeQueries([ApiEndpoints.DISCORD_CHANNELS], {
+        exact: true,
+      });
```

Comment:
We can't do this anymore. We are in the process of tidy up the mess so please align with new pattern.
Basically API calls should by done with react-query and all of these api related logic should be handled in `state/api` directory, following existing patterns. As @mzparacha worked recently in this area, let's make sure that it is done properly